utils/tinystr/src/ascii.rs

```diff
@@ -281,6 +281,67 @@ impl<const N: usize> TinyAsciiStr<N> {
         unsafe { TinyAsciiStr::from_utf8_unchecked(bytes) }
     }
 
+    #[inline]
+    #[must_use]
+    /// Returns a `TinyAsciiStr<Q>` with the concatenation of this string,
+    /// `TinyAsciiStr<N>`, and another string, `TinyAsciiStr<M>`.
+    ///
+    /// If `Q < N + M`, the string gets truncated.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tinystr::tinystr;
+    /// use tinystr::TinyAsciiStr;
+    ///
+    /// let abc = tinystr!(6, "abc");
+    /// let defg = tinystr!(6, "defg");
+    ///
+    /// // The concatenation is successful if Q is large enough...
+    /// assert_eq!(
+    ///     abc.concat(defg),
+    ///     tinystr!(16, "abcdefg")
+    /// );
+    /// assert_eq!(
+    ///     abc.concat(defg),
+    ///     tinystr!(12, "abcdefg")
+    /// );
+    /// assert_eq!(
+    ///     abc.concat(defg),
+    ///     tinystr!(8, "abcdefg")
+    /// );
+    /// assert_eq!(
+    ///     abc.concat(defg),
+    ///     tinystr!(7, "abcdefg")
+    /// );
+    ///
+    /// /// ...but it truncates of Q is too small.
+    /// assert_eq!(
+    ///     abc.concat(defg),
+    ///     tinystr!(6, "abcdef")
+    /// );
+    /// assert_eq!(
+    ///     abc.concat(defg),
+    ///     tinystr!(2, "ab")
+    /// );
+    /// ```
+    pub const fn concat<const M: usize, const Q: usize>(
+        self,
+        other: TinyAsciiStr<M>,
+    ) -> TinyAsciiStr<Q> {
```

Comment:
`nit`: I think you should debug assert that Q = N+M.

Reply:
Did you see my comment above about why I didn't add the assertion? It's fine if you disagree and want me to add the assertion, but I wanted to verify that you read my reasoning and are making this comment despite it

Reply:
Please also note that the only use case for this function will involve `.concat().resize()` if I force `Q` to be `N + M`

Reply:
I don't think I understand your logic. My interpretation of your code is that I can now do:

```
let t: TinyAsciiStr<12> = tinystr!(4, "abcd").concat(tinystr!(4, "efgh"));
```